src/cmd_line/commands/write.ts

```diff
@@ -56,27 +56,26 @@ export class WriteCommand extends node.CommandBase {
       throw error.VimError.fromCode(error.ErrorCode.E32);
     }
 
-    fs.access(this.activeTextEditor.document.fileName, fs.W_OK, (accessErr) => {
-      if (accessErr) {
-        if (this.arguments.bang) {
-          fs.chmod(this.activeTextEditor.document.fileName, 666, (e) => {
-            if (e) {
-              modeHandler.setupStatusBarItem(e.message);
-            } else {
-              this.save(modeHandler);
-            }
-          });
-        } else {
-          modeHandler.setupStatusBarItem(accessErr.message);
-        }
+    try {
+      fs.accessSync(this.activeTextEditor.document.fileName, fs.W_OK);
```

Comment:
Hey can you help me understand why we use `sync` instead of `async` methods here?